### PR TITLE
[flatbuffers] upgrade the `flatbuffers` to v2.0.0

### PR DESCRIPTION
**Pull request by jixingcn**
**Describe the pull request**

- #### What does your PR fix?  
  Upgrade the `flatbuffers` to v2.0.0

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
  Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/flatbuffers/CONTROL [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Source: flatbuffers
-Version: 1.12.0
-Port-Version: 2
+Version: 2.0.0
+Port-Version: 0
```
Comment (JonLiu1993) [inline ports/flatbuffers/CONTROL:3]:
```suggestion

```

File: ports/mnn/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "mnn",
   "version-string": "1.1.0",
```
Comment (JonLiu1993) [inline ports/mnn/vcpkg.json:3]:
```suggestion
  "version": "1.1.0",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@jixingcn ,Thanks for your pr.
have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？

**Comment by jixingcn** [comment]
> @jixingcn ,Thanks for your pr.
> have you tested the feature locally on x64-linux, x64-windows, and x64-windows-static？
> 

My application use `flatbuffers` on x64-windows and x64-windiws-static. I think all features are tested or verified.

**Comment by mathisloge** [comment]
@JonLiu1993 @jixingcn it does seem a bit suspicious to add a 19000 lines patch file. 
What is the context of this patch. Shouldn't it be a upstream change?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 @jixingcn it does seem a bit suspicious to add a 19000 lines patch file.
> What is the context of this patch. Shouldn't it be a upstream change?

Sorry, I did not find a patch file with 19000 lines. Can you point it out for me? Thank you

**Comment by JonLiu1993** [comment]
@jixingcn,Please run the command "./vcpkg x-add-version flatbuffers --overwrite-version" and commit the changes again.

**Comment by jixingcn** [comment]
> @jixingcn,Please run the command "./vcpkg x-add-version flatbuffers --overwrite-version" and commit the changes again.

I already run, but nothing's changed.

**Comment by jixingcn** [comment]
> > @JonLiu1993 @jixingcn it does seem a bit suspicious to add a 19000 lines patch file.
> > What is the context of this patch. Shouldn't it be a upstream change?
> 
> Sorry, I did not find a patch file with 19000 lines. Can you point it out for me? Thank you

You can find in [https://github.com/microsoft/vcpkg/pull/18897/commits/96e57b8a0b29e2491abdfa7f6871f0dbf2bed3a0](https://github.com/microsoft/vcpkg/pull/18897/commits/96e57b8a0b29e2491abdfa7f6871f0dbf2bed3a0).

Commit this patch because [the pipeline's report](https://dev.azure.com/vcpkg/public/_build/results?buildId=55891&view=results). The 19000+ lines were generated by [MNN's schema generate script](https://github.com/alibaba/MNN/blob/master/schema/generate.ps1) and [the latest `flatc`](https://google.github.io/flatbuffers/flatbuffers_guide_tutorial.html) that was compiled by `vcpkg`.

**Review by JonLiu1993 — APPROVED** [review]

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
We cannot commit a 19k line patch like this, especially since it will need to change every time flatbuffers updates -- instead, `mnn`'s port should run `flatc` at build time to generate any needed files.

You can easily get a runnable `flatc.exe` by depending on `flatbuffers` as a host dependency:

```json
{
  "dependencies": [
    { "host": true, "name": "flatbuffers" }
  ]
}
```

Then, in the portfile, you can use `${CURRENT_HOST_INSTALLED_DIR}/tools/...` to access the host dependency.

**Comment by jixingcn** [comment]
> We cannot commit a 19k line patch like this, especially since it will need to change every time flatbuffers updates -- instead, `mnn`'s port should run `flatc` at build time to generate any needed files.
> 
> You can easily get a runnable `flatc.exe` by depending on `flatbuffers` as a host dependency:
> 
> ```json
> {
>   "dependencies": [
>     { "host": true, "name": "flatbuffers" }
>   ]
> }
> ```
> 
> Then, in the portfile, you can use `${CURRENT_HOST_INSTALLED_DIR}/tools/...` to access the host dependency.

OK, thanks. I will fix that.

File: ports/mnn/vcpkg.json [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
   "name": "mnn",
-  "version-string": "1.1.0",
+  "version": "1.1.0",
+  "port-version": 1,
   "description": "MNN is a blazing fast, lightweight deep learning framework, battle-tested by business-critical use cases in Alibaba",
   "homepage": "https://www.mnn.zone/",
   "license": "Apache-2.0",
   "supports": "!(arm | uwp | ios | android)",
   "dependencies": [
-    "flatbuffers",
```
Comment (mathisloge) [inline ports/mnn/vcpkg.json:9]: doesn't mnn needs flatbuffers at runtime, too?

**Comment by JonLiu1993** [comment]
@jixingcn ,Could you please take a look?
https://dev.azure.com/vcpkg/public/_build/results?buildId=56048&view=artifacts&pathAsName=false&type=publishedArtifacts

File: ports/mnn/vcpkg.json (hunk 3 above)
Comment (jixingcn) [reply ports/mnn/vcpkg.json:9]: Sorry, I am wrong. I think `{ "host": true, "name": "flatbuffers" }` equal `"flatbuffers"` before. I am fixing.

**Comment by jixingcn** [comment]
> @jixingcn ,Could you please take a look?
> https://dev.azure.com/vcpkg/public/_build/results?buildId=56048&view=artifacts&pathAsName=false&type=publishedArtifacts

Ok

File: ports/mnn/vcpkg.json (hunk 3 above)
Comment (jixingcn) [reply ports/mnn/vcpkg.json:9]: Committed those after built it on my computer, but no error.

Comment (jixingcn) [reply ports/mnn/vcpkg.json:9]: Thanks a lot.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/mnn/vcpkg.json (hunk 3 above)
Comment (mathisloge) [reply ports/mnn/vcpkg.json:9]: It will only affect build, if you crosscompile.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@jixingcn ,Thanks for your contribution.

**Comment by jixingcn** [comment]
> @jixingcn ,Thanks for your contribution.

Thanks for your patient. 😃

**Pull request merged**